metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -936,21 +944,13 @@ private QuorumController(LogContext logContext,
         this.replicationControl = new ReplicationControlManager(snapshotRegistry,
             logContext, defaultReplicationFactor, defaultNumPartitions,
             configurationControl, clusterControl, controllerMetrics);
-        this.logManager = logManager;
+        this.raftClient = raftClient;
         this.metaLogListener = new QuorumMetaLogListener();
-        this.curClaimEpoch = -1L;
-        this.lastCommittedOffset = snapshotReader.epoch();
+        this.curClaimEpoch = -1;
+        this.lastCommittedOffset = -1L;
         this.writeOffset = -1L;
 
-        while (snapshotReader.hasNext()) {
-            List<ApiMessage> batch = snapshotReader.next();
-            long index = 0;
-            for (ApiMessage message : batch) {
-                replay(message, snapshotReader.epoch(), index++);
-            }
-        }
-        snapshotRegistry.createSnapshot(lastCommittedOffset);
```

Comment:
To minimize the number of changes needed to merge `trunk` into this PR, I removed the code that was loading snapshots. This is okay because the quorum controllers are not generating snapshots so there aren't any snapshots to load.

I will add snapshot loading and generating on the controller as part of https://issues.apache.org/jira/browse/KAFKA-12787

Reply:
Hmm... I would rather not remove this code if possible. In another comment I suggested moving it to the snapshot load function that you also added. That would also allow the associated test to keep working.

Reply:
Okay. I didn't want do this as part of the merge commit but I'll add it back as a separate commit.

Reply:
Done. Added snapshot loading at startup and added back the test for it.

There is some code duplication in the tests that I'll remove as part of https://issues.apache.org/jira/browse/KAFKA-12787